src/lib.rs

```diff
@@ -335,11 +337,13 @@ impl<'p, 'a> Serialize for SerializePyObject<'p, 'a> {
                 for (key, value) in x {
                     if key == self.py.None().as_ref(self.py) {
                         map.serialize_key("null")?;
-                    } else if let Ok(key) = key.extract::<bool>() {
-                        map.serialize_key(if key { "true" } else { "false" })?;
                     } else if let Ok(key) = key.str() {
                         let key = key.to_string().map_err(debug_py_err)?;
-                        map.serialize_key(&key)?;
+                        match key {
+                            Cow::Borrowed("true") => map.serialize_key(&true)?,
```

Comment:
I don't think this branch is needed -- a boolean key `true` and a string key `"true"` serialize exactly the same in JSON.

Reply:
Just to be clear, you mean we could get rid of the match for `Cow::Borrowed("true")` and `Cow::Borrowed("here")` and simply always use the `_` case?

Reply:
Seems to work in the unit tests at least. 😅 Let me push that change.

Reply:
Yes.